raw-spans-grouper/src/main/java/org/hypertrace/core/rawspansgrouper/RawSpanToStructuredTraceGroupingJob.java

```diff
@@ -120,6 +123,12 @@ public void run() throws Exception {
   public void stop() {
   }
 
+  @VisibleForTesting
```

Comment:
I hate this - can we mock `FlinkUtils.getExecutionEnvironment` ? Or is that when you get in to all the powermock shenanigans?

Reply:
I do too. You'd have to pass in the environment into the constructor.

Reply:
So either way you have a package private method to access the environment.

Reply:
Not sure about what's mockable in a static context, but if `getExecutionEnvironment` can be mocked, you can have it return a mock that you then verify. At least that's how it works in the non-static context.

Reply:
Yeah. But that means using power-mock which is a no-no.

Reply:
Got it - that's what I was getting at with the static questions :)